hack/update-submodule.sh

```diff
@@ -0,0 +1,92 @@
+
+set -euo pipefail
+
+function help() {
+    echo "Usage: update_submodules.sh [OPTIONS]... branch_name..."
+    echo "Update submodules HEAD to remote branches"
+    echo "Must be ran in repo root directory"
+    echo ""
+    echo "Options:"
+    echo "-m   Specific modules to update, separated by spaces. Defaults to all submodules"
+    echo "-h   Shows usage text"
+    echo "Example:"
+    echo "update_submodules.sh master release-4.6"
+    echo "update_submodules.sh master -m \"kubernetes ovn-kubernetes\""
+}
+
+# Stages changes to the given submodule and makes a commit including those changes
+function generate_submodule_commit() {
+    local submodule=$1
+    cd $submodule
+    origin_url=$(git remote get-url origin)
+    short_head=$(git rev-parse --short HEAD)
+    long_head=$(git rev-parse HEAD)
+    cd ..
+    git add $submodule
+    # Commit changes if there are any
+    if git commit -m "[submodule][$submodule] Update to $short_head" -m "Bump to $origin_url/commit/$long_head" -m "This commit was generated using hack/update_submodules.sh"; then
+        echo "New commit for $submodule"
+    fi
+}
+
+# Creates a new branch based off the given branch with commits updating all submodules provided in a space separated list
+function update_submodules_for_branch() {
+    local base_branch=$1
+    local new_branch=$2
+    local modulelist=$3
+
+    if git branch -D $new_branch; then
+	echo "Deleted branch $new_branch"
+    fi
+    git checkout $base_branch -b $new_branch
+
+    # Generate a commit updating each submodule
+    for submodule in $modulelist; do
+        git submodule update --remote $submodule
+        generate_submodule_commit $submodule
+    done
+}
+
+modulelist=""
+while getopts "m:h?" opt; do
+    case "$opt" in
+    m) modulelist=$OPTARG;;
+    h|\?) help; exit 0;;
+    esac
+done
+shift $((OPTIND -1))
+
+# If the -m option wasnt given, update all modules
+if [ -z "$modulelist" ]; then
+        modulelist=$(git config --file .gitmodules --get-regexp path | awk '{ print $2 }')
+fi
+
+# Return if the working branch is dirty
+if ! git diff --quiet; then
+  echo "branch dirty, exiting to not overwrite work"
+  exit 1
+fi
+
+# For each branch that the user gives (master, release-4.7), create a new branch based off it, with commits updating the submodules
+initial_branch=$(git branch --show-current)
+new_branch_list=()
+for branch in "$@"; do
+    new_branch="$branch-submodule-update-$(date +%m-%d)"
+    update_submodules_for_branch $branch $new_branch $modulelist
+    new_branch_list+=($new_branch)
+done
+
+# Return to initial branch
+git checkout $initial_branch
+echo "****"
+echo "New branches created:"
+echo "    [${new_branch_list[@]}]"
+echo ""
+echo "For each branch, you may push to your fork and create a PR against openshift/windows-machine-config-operator"
+echo "example:"
+echo "# assumes you have your fork set as the remote 'origin'"
+echo "git push origin ${new_branch_list[0]}"
```

Comment:
Will this push to all the branches?

Reply:
No this is just giving an example for one of the brances